THEMES.md

```diff
@@ -0,0 +1,63 @@
+If you're exploring this repository without having sniffed a specific smell,
+feel free to explore the examples by themes.
+
+## Subjects lack focus and intention in their responsibility
+Work through the following examples, giving careful thought to the concepts of
+Single Responsibility Principle. Subjects should have one purpose, rather that's
+calculation or collaboration. When those two are mixed, tests become hard to
+write and understand.
+
+*Missing Assertions*: Subject does too much, making edges hard to test.
+*Bury the Lede*: Setup too complex.
+*Chafing*: Test pain -> Test design instead of code design.
+*Tangential*: Mixing levels of abstractions / types of responsibilities.
+*7 Layer Testing*: Test can't escape interaction with God object.
+*Contaminated Test Subject*: Very large or stateful subject.
+*Invasion of Privacy*: Testing code that should be it's own unit.
+*X-Ray Specs*: Tests secretly edit private state. Subject has too many responsibilities.
+
+## Tests aren't isolated and are testing more than just a single unit
+Work through the following examples, giving careful thought to role of the test
+within the test suite, being sure to consider what experiment the test is conducting.
+
+*Tangential*: Testing subject depends on unrelated things.
+*7 Layer Testing*: Testing subject depends on "reusable blocks" below subject.
+*Complex Assertions*: Test relies on side effects.
+*Indecisive*: Inconsistent setup causes tests to check data's state.
+*Premature Assertions*: Lack of confidence in subject's dependencies
+*Mockers without Borders*: Some dependencies are polluting the test, some are mocked.
+
+## Testing too much
+Work through the following examples, analyzing the costs and benefits of each test.
+
+*Invisible Assertions*: Sky might fall / Doesn't blow up.
+*Quixotic*: Overly integrated journey.
+*Long*: Testing multiple things.
+*Generative*: More than necessary.
```

Comment:
Testing numerous redundant examples, watering down the degree to which the test expresses your intention